graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -460,10 +518,23 @@ abstract class BaseDataTypeGenerator(
                 .addMethod(buildMethod)
 
         javaType.build().fieldSpecs.map {
-            MethodSpec.methodBuilder(it.name)
+            val method = MethodSpec.methodBuilder(it.name)
                 .addJavadoc(it.javadoc)
                 .returns(builderClassName)
                 .addStatement("this.${it.name} = ${it.name}")
+
+            val fieldName = it.name
+            val field = fields.find { it.name.contains(fieldName) }
```

Comment:
This should be  
```
val field = fields.find { iter-> ReservedKeywordSanitizer.sanitize(iter.name) == fieldName }
```

Otherwise it matches other fields that are similarly named and is incorrect.
e.g. `something` and `isSomethingThere`..will generate `isSomething` that conflicts with `isSomethingThere` field.

Also this is problematic if the field `isSomething` is also there as part of the schema.